src/control/RecentManager.cpp

```diff
@@ -101,6 +101,12 @@ auto RecentManager::sortRecentsEntries(GtkRecentInfo* a, GtkRecentInfo* b) -> gi
     return tp_a != tp_b ? (tp_a < tp_b ? 1 : -1) : 0;
 }
 
+auto RecentManager::getMostRecent() -> GtkRecentInfo* {
+    GList* filteredItemsXoj = filterRecent(gtk_recent_manager_get_items(gtk_recent_manager_get_default()), true);
+
+    return static_cast<GtkRecentInfo*>(filteredItemsXoj->data);
```

Comment:
I think this will leak memory.

Reply:
i'm not that used to c++. Would make sense to extend `filterRecent` to give it an amount/count of needed results (in this case 1) and then only return only the filepath and before returning destroying the single `GtkRecentInfo` object or which flow would you suggest to not leak memory in this case?

Reply:
In my opinion, the easiest change is to just free the other recent items and then keep the recent info that we want.
1. Store the `GtkRecentInfo*` we want in a local variable, and then set the corresponding data pointer of the GList to null.
2. Call [`g_free_list_full`](https://developer.gnome.org/glib/stable/glib-Doubly-Linked-Lists.html#g-list-free-full) on `filteredItemsXoj`
3. Return the saved `GtkRecentInfo*`.

Reply:
This is really more like C rather than C++, since glib and GTK are C libraries. See https://developer.gnome.org/programming-guidelines/stable/memory-management.html.en

Reply:
hopefully i changed everything as it should be.

i'm more used to "simple" languages like php, javascript and ruby. :)